### DISCUSSION TITLE
How I can get the CDN Feature to work?

### DISCUSSION THREAD

**Discussion by DunklerPhoenix** [photoprism/photoprism#3358] [Q&A]
Heho
I registered and account at [bunny](https://docs.photoprism.app/getting-started/using-a-cdn/#bunnynet) and set up the ENV like this:

```
      PHOTOPRISM_CDN_URL: "https://cdn.domain.com"
      PHOTOPRISM_CDN_VIDEO: "true"
      PHOTOPRISM_HTTP_CACHE_MAXAGE: 2592000
      PHOTOPRISM_HTTP_CACHE_PUBLIC: "true"
```

Photoprism uses them, but chrome refuses to load the content:

`Refused to load the image '<URL>' because it violates the following Content Security Policy directive: "default-src 'self' 'unsafe-inline' data: blob: wss: *.photoprism.app *.maptiler.com". Note that 'img-src' was not explicitly set, so 'default-src' is used as a fallback.
`

Could this be a problem with my traefik or did I forgot a setting?

**Comment by lastzero** [comment]
Thanks for bringing this up!

The problem likely comes from the default [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP). If you are using our Plus version, you can change it using the documented config option:

https://docs.photoprism.app/getting-started/config-options/#web-server

Alternatively, you can change the header in the HTTPS reverse proxy you are using, for example in Traefik:

https://docs.photoprism.app/getting-started/proxies/traefik/

In future releases, we will try to automatically update the CSP when a CDN is configured, and will also make the CSP config option available in our Community Edition.

**Reply by lastzero** [reply]
I've created a follow-up issue for this:
- https://github.com/photoprism/photoprism/issues/3367

**Reply by DunklerPhoenix** [reply]
Thank you.
I set it to this Value:

`      PHOTOPRISM_HTTP_CSP: "default-src 'self' cdn.domain.com; img-src 'self' cdn.domain.com; media-src 'self' cdn.domain.com; script-src 'self' 'unsafe-inline' cdn.domain.com; style-src 'unsafe-inline' 'self' cdn.domain.com"`


I need to use 'unsafe-inline' because some scripts are directly added and not included via file.
But this works.

**Reply by lastzero** [reply]
Excellent! Glad you got it working. Web security should be better covered in our docs.

**Reply by DunklerPhoenix** [reply]
It would be also great if the Environment Variables have some examples.

At first i thought PHOTOPRISM_CDN_VIDEO also needs to be an url. 
Its really difficult to figure out the correct values

**Reply by lastzero** [reply]
It might be more obvious when you use the built-in commands eg run `photoprism --help` in a terminal to see all options and `photoprism config` to see the current values.

**Comment by pgalbavy** [comment]
As a late follow-up, the documentation is unclear. The CDN URL - what is it? Is it the externally accessible URL of your site via the CDN or some special URL for the CDN itself? I have set-up Cloudflare on a personal account and it seems to work, but I would like to do it right.

If I do not set the CDN_URL, can I still set CDN_VIDEO to true or is it dependent on the URL being set to something?

Thanks!

**Reply by lastzero** [reply]
If you configure CloudFlare as a web application firewall / proxy, you won't need any of this. Note however that their free service does not include video streaming.

**Reply by pgalbavy** [reply]
Thanks for confirming. It's more for when I travel and/or share albums with others via links - I have a divert at home :)
